blockdag/dag.go

```diff
@@ -1058,21 +1058,24 @@ func (dag *BlockDAG) meldVirtualUTXO(newVirtualUTXODiffSet *DiffUTXOSet) error {
 	return newVirtualUTXODiffSet.meldToBase()
 }
 
-func (node *blockNode) diffFromTxs(pastUTXO UTXOSet, transactions []*util.Tx) (*UTXODiff, error) {
-	diff := NewUTXODiff()
+func (node *blockNode) calcUTXO(pastUTXO *DiffUTXOSet, transactions []*util.Tx) (UTXOSet, error) {
+	utxo := NewDiffUTXOSet(pastUTXO.base, pastUTXO.UTXODiff.clone())
 
 	for _, tx := range transactions {
-		txDiff, err := pastUTXO.diffFromTx(tx.MsgTx(), UnacceptedBlueScore)
+		isAccepted, err := utxo.AddTx(tx.MsgTx(), UnacceptedBlueScore)
 		if err != nil {
 			return nil, err
 		}
-		diff, err = diff.WithDiff(txDiff)
-		if err != nil {
-			return nil, err
+
+		if !isAccepted {
+			return nil, ruleError(ErrMissingTxOut, fmt.Sprintf(
+				"Transaction %s is invalid because because at least one of its inputs doesn't have a "+
```

Comment:
Instead of saying "because at least one of its inputs...", I'd rather have the input's outpoint specified explicitly